html/js/tree/Tree.js

```diff
@@ -0,0 +1,195 @@
+        // In case there is no arguments start from root
+        if (node === undefined) { node = this.root; }
+
+		if (!node.isLeaf()) {
+            for (var i = 0; i < node.children.length; i++) {
+                this.drawLineBetweenNodes(node, node.children[i]);
+                this.recursiveDraw(node.children[i]);
+            }
+        }
+        node.draw();
+	};
+
+	/**
+	* Function that draws a line between two given nodes
+	* @param {Node} node1 Node A
+	* @param {Node}	node2 Node B
+	*/
+    Tree.prototype.drawLineBetweenNodes = function(node1, node2){
+        var circleRadius = GAMBIT.CONSTANTS.CIRCLE_SIZE/2;
+        GAMBIT.canvas.line(node1.x + circleRadius, node1.y + circleRadius, node2.x + circleRadius, node2.y +circleRadius).stroke({ width: 1 });
+    };
+
+	/**
+	* Function that returns the number of leaves in the tree
+	* @return {Number} leafCounter Number of Leaves
+	*/
+    Tree.prototype.numberLeaves = function () {
+        this.leaves = [];
+        return this.recursiveNumberLeaves(this.root);
+    };
+
+
+	/**
+	* Recursive function that returns the number of leaves below a determinate node
+	* Leaves positions are updated in a different function.
+	* Stopping criteria: that the current node is a leaf
+	* Recursive expansion: to all of the node's children
+	* @param	{Node}		node 		Node to start from
+	* @return	{Number}	leafCounter Number of Leaves
+	*/
+    Tree.prototype.recursiveNumberLeaves = function (node) {
+        if (node.isLeaf()) {
+            this.leaves.push(node);
+            return 1;
+        } else {
+            var leafCounter = 0;
+            for (var i = 0; i < node.children.length; i++) {
+                leafCounter += this.recursiveNumberLeaves(node.children[i]);
+            }
+            return leafCounter;
+        }
+    };
+
+	/**
+	* Function that updates the positions of the nodes in the tree
+	*/
+    Tree.prototype.updatePositions = function () {
+        this.updateLeavesPositions();
+        this.recursiveUpdatePositions(this.root);
+        this.positionsUpdated = true;
+    };
+
+	/**
+	* Recursive function that updates the positions of the node children.
+	* Leaves positions are updated in a different function.
+	* Stopping criteria: that the current node is a leaf
+	* Recursive expansion: to all of the node's children
+	* @param {Node} node Node that will get their children's positions updated
+	*/
+    Tree.prototype.recursiveUpdatePositions = function (node) {
+        if (!node.isLeaf()) {
+            for (var i = 0; i < node.children.length; i++) {
+                this.recursiveUpdatePositions(node.children[i]);
+            }
+            // Get middle point between the children most in the left and most
+			// in the right
+            // TODO: apply level weighted function for special cases
+            node.x = node.children[0].x +
+				(node.children[node.children.length-1].x - node.children[0].x)/2;
+            node.y = node.level * 50;
```

Comment:
50 should be externalized to CONSTANTS (e.g., DIST_BETWEEN_LEVELS)
 Add a line note